core/node/libp2p/smux.go

```diff
@@ -3,52 +3,25 @@ package libp2p
 import (
 	"fmt"
 	"os"
-	"strings"
 
 	"github.com/ipfs/kubo/config"
 
 	"github.com/libp2p/go-libp2p"
-	"github.com/libp2p/go-libp2p/p2p/muxer/mplex"
 	"github.com/libp2p/go-libp2p/p2p/muxer/yamux"
 )
 
 func makeSmuxTransportOption(tptConfig config.Transports) (libp2p.Option, error) {
 	if prefs := os.Getenv("LIBP2P_MUX_PREFS"); prefs != "" {
-		// Using legacy LIBP2P_MUX_PREFS variable.
-		log.Error("LIBP2P_MUX_PREFS is now deprecated.")
-		log.Error("Use the `Swarm.Transports.Multiplexers' config field.")
-		muxers := strings.Fields(prefs)
-		enabled := make(map[string]bool, len(muxers))
-
-		var opts []libp2p.Option
-		for _, tpt := range muxers {
-			if enabled[tpt] {
-				return nil, fmt.Errorf(
-					"duplicate muxer found in LIBP2P_MUX_PREFS: %s",
-					tpt,
-				)
-			}
-			switch tpt {
-			case yamux.ID:
-				opts = append(opts, libp2p.Muxer(tpt, yamux.DefaultTransport))
-			case mplex.ID:
-				opts = append(opts, libp2p.Muxer(tpt, mplex.DefaultTransport))
-			default:
-				return nil, fmt.Errorf("unknown muxer: %s", tpt)
-			}
-		}
-		return libp2p.ChainOptions(opts...), nil
-	} else {
-		return prioritizeOptions([]priorityOption{{
-			priority:        tptConfig.Multiplexers.Yamux,
-			defaultPriority: 100,
-			opt:             libp2p.Muxer(yamux.ID, yamux.DefaultTransport),
-		}, {
-			priority:        tptConfig.Multiplexers.Mplex,
-			defaultPriority: config.Disabled,
-			opt:             libp2p.Muxer(mplex.ID, mplex.DefaultTransport),
-		}}), nil
+		return nil, fmt.Errorf("configuring muxers with LIBP2P_MUX_PREFS is no longer supported")
+	}
+	if tptConfig.Multiplexers.Mplex != 0 {
+		return nil, fmt.Errorf("Swarm.Transports.Multiplexers.Mplex is no longer supported")
```

Comment:
If someone explicitly enabled it at some point and forgot, let's give people a pointer how to fix and why:
```suggestion
		return nil, fmt.Errorf("Swarm.Transports.Multiplexers.Mplex is no longer supported, remove it from your config, see https://github.com/libp2p/specs/issues/553")
```